file_store/src/file_sink.rs

```diff
@@ -384,6 +407,28 @@ impl FileSink {
     }
 }
 
+/// List of all files written by a file sink
+#[derive(Clone, Debug, Default)]
+pub struct Manifest {
+    written_files: Arc<Mutex<Vec<String>>>,
```

Comment:
We could avoid the `Arc<Mutex` if we retrieve the manifest list by sending a message to the file_sink, which then could send the results back in a oneshot.  This should also mean we don't need the flush message to send anything back on a oneshot.